src/sdk.spec.js

```diff
@@ -100,11 +100,10 @@ describe('OpenForm', () => {
       lang: 'nl',
     });
 
-    await act(async () => {
-      await form.init();
-      await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
-    });
+    await act(async () => await form.init());
 
+    // wait for the loader to be removed when all network requests have completed
+    await waitForElementToBeRemoved(() => within(formRoot).getByRole('status'));
```

Comment:
I don't understand this. Does the `init` promise resolve before api calls are completed?

Reply:
I think something changes with the `createRoot(...)` and then the `render` after of React 18.

In general, dealing with the `useAsync` effects is nasty in this upgrade, there are many warnings about state updates happening outside of the `act` calls, but I spent 2 hours trying to fix/understand them and I just don't see how it can reasonably be done. I think the proper path forward here is to get away from the `react-dom/test-utils` and instead use `testing-library/react` which is much more interaction focused anyway.

In general I find the `act` usage confusing as **** and it's one of the reasons I prefer writing storybook interaction tests :grimacing:

Reply:
Could it be the way we setup and teardown?  I don't know.. Ship it. :wink: